crates/autopilot/src/database/onchain_order_events/ethflow_events.rs

```diff
@@ -94,31 +116,46 @@ impl OnchainOrderParsing<EthFlowData, EthFlowDataForDb> for EthFlowOnchainOrderP
             .context("append_ethflow_orders failed during appending pre_interactions")
     }
 
-    fn customized_event_data_for_event_index(
+    async fn customized_event_data_for_event_index(
         &self,
         event_index: &EventIndex,
+        quote: &shared::order_quoting::Quote,
         order: &Order,
         hashmap: &HashMap<EventIndex, EthFlowData>,
         _onchain_order_placement: &OnchainOrderPlacement,
-    ) -> EthFlowDataForDb {
-        EthFlowDataForDb {
+    ) -> Result<EthFlowDataForDb> {
+        let slippage = BigRational::new(
+            quote.data.quoted_buy_amount.to_big_int(),
+            order.buy_amount.to_bigint().unwrap(),
+        )
+        .to_f64()
+        .unwrap_or(f64::NAN);
+        let unix_timestamp_of_block = self
+            .get_unix_timestamp_of_block(event_index.block_number)
+            .await?;
```

Comment:
This means we are making a request to the node to get the block information every time this function is called. This happens for every event. I think we get events per block already so it would make more sense to get the block information once for all events of the same block.

In general it seems wrong to do expensive work here because this happens every time events change, right? 
(I don't understand the ethflow infrastructure that well so I'm not sure. Could be nice to have design documentation of it in some place that mentions all the moving parts that are involved that are hard to identify through the code, like when one components write to the db and another reads.)
For example, before Dusan optimized the event updater we would reinsert all the events of the last N blocks on every new block. This would result in a lot of calls to this function so it probably shouldn't be expensive. One solution is a cache for the block timestamps.

Reply:
> I think we get events per block already so it would make more sense to get the block information once for all events of the same block.

I agree in general. Though I made this trade-off on purpose: it is very unlikely that two ethflow orders are placed onchain exactly at the same block. In these single cases, I think it's fine to call it twice.

>For example, before Dusan optimized the event updater we would reinsert all the events of the last N blocks on every new block. 
Agreed.

>One solution is a cache for the block timestamps.
I thought it would be overkill for something that is likely not called twice. But you are structurally right. Implementing a cache is not too much work. Let me give it a try.

The documentation for the refunder part is here: 
https://github.com/cowprotocol/services/issues/626
( I admit that it could be more detailed)
and the general spec was defined here:
https://app.zenhub.com/workspaces/research--development-62f2313261703300136858f4/issues/cowprotocol/services/330
(though its a little bit outdated). If you have concrete questions, don't hesitate to DM me.